use-timescale/migration/dual-write-and-backfill.md

```diff
@@ -0,0 +1,476 @@
+---
+title: Low-downtime migrations with dual-write and backfill
+excerpt: Migrate a hypertable or entire database with low downtime
+products: [cloud, self_hosted]
+keywords: [backups, restore]
+tags: [recovery, logical backup, pg_dump, pg_restore]
+---
+
+# Dual-write and backfill
+
+Dual-write and backfill is a migration strategy to move a large amount of
+time-series data (100&nbsp;GB-4&nbsp;TB+) with low downtime (on the order of 
+minutes).
+Roughly, it consists of three steps:
+
+1. Clone schema and relational data from source to target
+1. Dual-write to source and target
+1. Backfill time-series data
+
+Dual-write and backfill can be used for any source database type, as long as it
+can provide data in csv format. It can be used to move data from a PostgresSQL
+source, and from TimescaleDB to TimescaleDB. If the source and target databases
+are PostgresSQL, they can be of different versions, as long as the target is
+greater than the source. If both source and target use TimescaleDB, the version
+of TimescaleDB must be the same.
+
+Dual-write and backfill works well when:
+1. The bulk of the (on-disk) data is in time-series tables.
+1. Writes by the application do not reference historical time-series data.
+1. There is no requirement for transactional consistency (that is, it is possible
+   to filter the time-series data by time and retain data integrity).
+1. No `UPDATE` or `DELETE` queries will be run on time-series data in the
+   source database during the migration process (or if they are, it happens in
+   a controlled manner, such that it's possible to either ignore, or
+   re-backfill).
+1. Either the relational (non-time-series) data is small enough to be copied
+   from source to target in an acceptable amount of time for this to be done
+   with downtime, or the relational data can be copied asynchronously while the
+   application continues to run (that is, changes relatively infrequently).
+
+## Migration process
+
+In detail, the migration process consists of the following steps:
+1. Set up a second database
+1. Modify the application to write to a secondary database
+1. Migrate schema and relational data from source to target
+1. Start the application in dual-write mode
+1. Determine the completion point `T`
+1. Backfill time-series data from source to target
+1. Enable retention and compression policies
```

Comment:
I think we should add a disable retention and compression policies in target database after `1. Migrate schema and relational data from source to target`

We talk about this in the `Set up schema and migrate relational data to new database`.

Reply:
IMHO, if we would do that, we would also need to split "Migrate schema and relational data from source to target" into two parts (this is effectively a table of contents for the stuff below). I'm hesitant to do that. WDYT?